tests/buyers/views/test_buyers.py

```diff
@@ -1812,79 +1801,129 @@ def test_page_shows_csv_download_link_if_brief_closed(self, data_api_client):
+            ]
         )
+        data_api_client.get_brief.return_value = api_stubs.brief(lot_slug="digital-outcomes")
 
-        assert res.status_code == 200
-        assert len(csv_link) == 0
-        assert "The file will be available here once applications have closed." in page
+        self.login_as_buyer()
+        res = self.client.get(
+            "/buyers/frameworks/digital-outcomes-and-specialists/requirements/digital-outcomes/1234/responses"
+        )
 
-    def test_404_if_brief_does_not_belong_to_buyer(self, data_api_client):
+        assert res.status_code == 404
+
+
+@mock.patch("app.buyers.views.buyers.data_api_client")
+class TestViewBriefResponsesPage(AbstractViewBriefResponsesPage):
+    two_good_three_bad_responses = {
+        "briefResponses": [
+            {"essentialRequirements": [True, True, True, True, True]},
+            {"essentialRequirements": [True, False, True, True, True]},
+            {"essentialRequirements": [True, True, False, False, True]},
+            {"essentialRequirements": [True, True, True, True, True]},
+            {"essentialRequirements": [True, True, True, True, False]},
+        ]
+    }
+
+    def test_page_shows_csv_download_link_if_brief_closed(self, data_api_client):
+        data_api_client.find_brief_responses.return_value = self.two_good_three_bad_responses
         data_api_client.get_framework.return_value = api_stubs.framework(
             slug='digital-outcomes-and-specialists',
             status='live',
             lots=[
                 api_stubs.lot(slug='digital-outcomes', allows_brief=True),
             ]
         )
-        data_api_client.get_brief.return_value = api_stubs.brief(lot_slug="digital-outcomes", user_id=234)
+        data_api_client.get_brief.return_value = api_stubs.brief(lot_slug="digital-outcomes", status='closed')
 
         self.login_as_buyer()
         res = self.client.get(
             "/buyers/frameworks/digital-outcomes-and-specialists/requirements/digital-outcomes/1234/responses"
         )
+        document = html.fromstring(res.get_data(as_text=True))
+        csv_link = document.xpath(
+            '//a[@href="/buyers/frameworks/digital-outcomes-and-specialists/requirements/digital-outcomes/1234/responses/download"]'  # noqa
+        )[0]
 
-        assert res.status_code == 404
+        assert res.status_code == 200
+        assert self._strip_whitespace(csv_link.text_content()) == \
+            "CSVdocument:Downloadsupplierresponsesto‘Ineedathingtodoathing’"
 
-    def test_404_if_lot_does_not_allow_brief(self, data_api_client):
+
+@mock.patch("app.buyers.views.buyers.data_api_client")
+class TestViewBriefResponsesPage2(AbstractViewBriefResponsesPage):
+    two_good_three_bad_responses = {
+        "briefResponses": [
+            {"essentialRequirementsMet": True, "essentialRequirements": [{"evidence": "blah"}]},
+            {"essentialRequirementsMet": False, "essentialRequirements": []},
```

Comment:
As spotted in your other pull request, this isn't actually expected data. You can't have a 'bad' response for the new flow because `essentialRequirementsMet` will either not be set (and therefore not returned by the API) or will be `True`.

Sorry! I was about to thumbs this up and all...